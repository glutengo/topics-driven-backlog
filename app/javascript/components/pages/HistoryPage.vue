<template>
    <section id="history-page">
        <h1>History</h1>

        <template v-for="sprint in sprints">
            <h2>{{ sprint.name }}</h2>

            <table v-if="storiesInSprint(sprint.id).length" class="table table-striped">
                <thead>
                    <tr>
                        <th>Identifier</th>
                        <th>Story</th>
                        <th>Story&nbsp;points</th>
                        <th>Status</th>
                    </tr>
                </thead>
                <tbody>
                    <tr is="story" v-for="story in storiesInSprint(sprint.id)" :key="story.id" :data="story" view="history"></tr>
                </tbody>
            </table>

            <p v-else>No stories in this sprint</p>
        </template>

        <h2>Backlog</h2>
        <table class="table table-striped">
            <thead>
                <tr>
                    <th>Identifier</th>
                    <th>Story</th>
                    <th>Story&nbsp;points</th>
                    <th>Status</th>
                </tr>
            </thead>
            <tbody>
                <tr is="story" v-for="story in backlog" :key="story.id" :data="story" view="history"></tr>
            </tbody>
        </table>
    </section>
</template>

<script>
    import Story from "../elements/Story.vue";
    import Sprint from "../elements/Sprint.vue";
    import * as _ from "lodash";
    import moment from "moment";

    export default {
        components: {Sprint, Story},
        data() {
            return {};
        },
        computed: {
            backlog() {
                return this.$store.getters['stories/all']
                    .filter(s => s.sprint_id === null)
                    .sort((a, b) => a.position - b.position);
            },

            sprints() {
                return this.$store.getters['courses/allSprints']()
                    .sort((a, b) => a.start_date.localeCompare(b.start_date));
            },
        },
        methods: {
            /**
             * All stories in a specific sprint
             * @param sprintId
             */
            storiesInSprint(sprintId) {
                return this.$store.getters['stories/find']('sprint_id', sprintId)
                    .sort((a, b) => a.position - b.position);
            },

            /**
             * Save current form state as new story
             */
            save() {
                this.$store.dispatch('stories/save', {
                    story: this.newStory
                }).then(() => {
                    this.newStory = {};
                });
            },

            /**
             * Save new position of dragged story
             * @param evt
             */
            saveOrder(evt) {
                const story = this.storiesInSprint[evt.oldIndex];

                if (!story) {
                    return;
                }

                this.$store.dispatch('stories/patch', {
                    id: story.id,
                    field: 'position',
                    value: evt.newIndex + 1, // act_as_list is 1-indexed
                    fetch: true,
                });
            },

            /**
             * Add story to next sprint
             * @param storyId
             */
            addToSprint(storyId) {
                if (!this.nextSprint) {
                    return;
                }

                this.$store.dispatch('stories/patch', {
                    id: storyId,
                    field: 'sprint_id',
                    value: this.nextSprint.id,
                });
            },

            /**
             * Remove story from next sprint
             * @param storyId
             */
            removeFromSprint(storyId) {
                this.$store.dispatch('stories/patch', {
                    id: storyId,
                    field: 'sprint_id',
                    value: null,
                });
            }
        }
    };
</script>

<style scoped>

</style>
