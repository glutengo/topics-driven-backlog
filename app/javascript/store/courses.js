import Vue from 'vue';

import { current } from '../helper/semester';

export default {
    namespaced: true,
    state: {
        initialised: false,
        data: []
    },
    mutations: {
        /**
         * Set one or all stories
         * @param state
         * @param payload
         */
        set: (state, payload) => {
            if (payload.courses) {
                Vue.set(state, 'data', payload.courses);
            }

            if (payload.course) {
                const isPresent = state.data
                    .find(course => course.id === payload.course.id);

                if (isPresent) {
                    Vue.set(state, 'data', state.data.map(course =>
                        (course.id === payload.course.id ? payload.course : course),
                    ));
                } else {
                    state.data.push(payload.course);
                }
            }
        },
    },
    actions: {
        /**
         * Initialises the courses store
         * @param state
         * @param dispatch
         */
        init({state, dispatch}) {
            if (!state.initialised) {
                dispatch('fetch').then(() => {
                    state.initialised = true;
                });
            }
        },

        /**
         * Fetches all courses from the API and updates store
         * @param commit
         * @returns {Promise}
         */
        fetch({commit}) {
            return new Promise((resolve, reject) => {
                Vue.http.get('/courses').then((response) => {
                    commit('set', {
                        courses: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Saves a course to the API and updates store
         * @param commit
         * @param payload (course)
         * @returns {Promise}
         */
        save({commit}, payload) {
            return new Promise((resolve, reject) => {
                Vue.http.post('/courses', payload.course).then((response) => {
                    commit('set', {
                        course: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Patches a given field of a given course with a given value
         * @param commit
         * @param payload (id, field, value)
         * @returns {Promise}
         */
        patch({commit}, payload) {
            return new Promise((resolve, reject) => {
                const data = {};
                data[payload.field] = payload.value;

                Vue.http.patch(`/courses/${payload.id}`, data).then((response) => {
                    commit('set', {
                        course: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },
    },
    getters: {
        /**
         * All courses
         * @param state
         */
        all: state => state.initialised
            ? state.data.sort((a, b) => a.position > b.position)
            : [],

        /**
         * A template for a new course
         * @returns {{semester: string}}
         */
        new: () => {
            return {
                semester: current().value,
            }
        }
    }
}