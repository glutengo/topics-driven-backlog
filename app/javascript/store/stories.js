import Vue from 'vue';

export default {
    namespaced: true,
    state: {
        initialised: false,
        statusMap: [
            {name: 'open', css: 'badge-dark'},
            {name: 'in progress', css: 'badge-warning'},
            {name: 'closed', css: 'badge-success'},
            {name: 'cancelled', css: 'badge-danger'}
        ],
        data: []
    },
    mutations: {
        /**
         * Set one or all stories
         * @param state
         * @param payload
         */
        set: (state, payload) => {
            if (payload.stories) {
                Vue.set(state, 'data', payload.stories);
            }

            if (payload.story) {
                const isPresent = state.data
                    .find(story => story.id === payload.story.id);

                if (isPresent) {
                    Vue.set(state, 'data', state.data.map(story =>
                        (story.id === payload.story.id ? payload.story : story),
                    ));
                } else {
                    state.data.push(payload.story);
                }
            }
        },
    },
    actions: {
        /**
         * Initialises the stories store
         * @param state
         * @param dispatch
         */
        init({state, dispatch}) {
            if (!state.initialised) {
                dispatch('fetch').then(() => {
                    state.initialised = true;
                });
            }
        },

        /**
         * Fetches all stories from the API and updates store
         * @param commit
         * @returns {Promise}
         */
        fetch({commit}) {
            return new Promise((resolve, reject) => {
                Vue.http.get('/stories').then((response) => {
                    commit('set', {
                        stories: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Saves a story to the API and updates store
         * @param commit
         * @param payload (story)
         * @returns {Promise}
         */
        save({commit}, payload) {
            return new Promise((resolve, reject) => {
                Vue.http.post('/stories', payload.story).then((response) => {
                    commit('set', {
                        story: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Patches a given field of a given story with a given value
         * @param commit
         * @param payload (id, field, value)
         * @returns {Promise}
         */
        patch({commit}, payload) {
            return new Promise((resolve, reject) => {
                const data = {};
                data[payload.field] = payload.value;

                Vue.http.patch(`/stories/${payload.id}`, data).then((response) => {
                    commit('set', {
                        story: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Saves order information to a story and triggers a new fetch of stories
         * @param getters
         * @param dispatch
         * @param payload
         * @returns {Promise}
         */
        reorder({getters, dispatch}, payload) {
            const story = getters.all[payload.oldIndex];

            return new Promise((resolve, reject) => {
                if (!story) {
                    return reject();
                }

                Vue.http.patch(`/stories/${story.id}`, {position: payload.newIndex}).then((response) => {
                    dispatch('fetch').then(resolve, reject);
                }, reject);
            });
        }
    },
    getters: {
        /**
         * All stories
         * @param state
         */
        all: state => state.initialised
            ? state.data.sort((a, b) => a.position > b.position)
            : [],

        /**
         * Find a story by identifier
         * @param state
         */
        byIdentifier: state => value => state.initialised
            ? state.data.find(s => s.identifier === value)
            : null,
    }
}