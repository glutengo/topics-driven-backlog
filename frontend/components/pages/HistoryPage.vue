<template>
    <section id="history-page">
        <template v-for="sprint in sprints">
            <div :key="sprint.id" class="mb-4">
                <h3 class="h5 text-muted mb-0">
                    <small>{{ sprint.start_date | displayDate }} - {{ sprint.end_date | displayDate }}</small>
                </h3>
                <h2>{{ sprint.name }}</h2>

                <table 
                    v-if="storiesInSprint(sprint.id).length" 
                    class="table table-striped">

                    <thead>
                        <tr>
                            <th>Identifier</th>
                            <th>Story</th>
                            <th>Topic</th>
                            <th>Story&nbsp;points</th>
                            <th>Status</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr 
                            is="StoryItem" 
                            v-for="story in storiesInSprint(sprint.id)" 
                            :key="story.id" 
                            :data="story" 
                            view="history"/>
                    </tbody>
                </table>

                <p v-else>No stories in this sprint</p>
            </div>
        </template>

        <h2>Backlog</h2>
        <table class="table table-striped">
            <thead>
                <tr>
                    <th>Identifier</th>
                    <th>Story</th>
                    <th>Topic</th>
                    <th>Story&nbsp;points</th>
                    <th>Status</th>
                </tr>
            </thead>
            <tbody>
                <tr 
                    is="StoryItem" 
                    v-for="story in backlog" 
                    :key="story.id" 
                    :data="story" 
                    view="history"/>
            </tbody>
        </table>
    </section>
</template>

<script>
import BaseProjectAwarePage from '@/components/pages/BaseProjectAwarePage';
import StoryItem from '@/components/elements/StoryItem';

export default {
    components: { StoryItem },
    extends: BaseProjectAwarePage,
    data() {
        return {};
    },
    computed: {
        /**
         * All stories not in sprints, sorted by position
         * @returns {array}
         */
        backlog() {
            return this.project
                ? this.$store.getters['stories/all'](this.project.id)
                    .filter(s => !s.sprint_id)
                    .sort((a, b) => a.position - b.position)
                : [];
        },

        /**
         * All sprints, sorted by start date
         * @returns {array}
         */
        sprints() {
            return this.course
                ? this.$store.getters['sprints/all'](this.course.id)
                    .sort((a, b) => a.start_date.localeCompare(b.start_date))
                : [];
        },
    },
    methods: {
        /**
         * All stories in a specific sprint
         * @param {int} sprintId
         * @returns {array}
         */
        storiesInSprint(sprintId) {
            return this.project
                ? this.$store.getters['stories/find'](this.project.id, 'sprint_id', sprintId)
                    .sort((a, b) => a.position - b.position)
                : [];
        },
    },
};
</script>
