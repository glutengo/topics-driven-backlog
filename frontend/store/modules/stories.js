import { generate } from '@/store/base/resource-module/index';
import resources from '@/store/resources';

const resourceModule = generate({
    resource: resources.STORIES,
    parent: resources.PROJECTS,
    state: {
        STATUS: { OPEN: 0, IN_PROGRESS: 1, CLOSED: 2, CANCELLED: 3 },
        statusMap: [
            { name: 'open', css: 'badge-dark' },
            { name: 'in progress', css: 'badge-warning' },
            { name: 'closed', css: 'badge-success' },
            { name: 'cancelled', css: 'badge-danger' },
        ],
    },
    template: () => {
        return {
            topic_id: null,
        };
    },
});

export default resourceModule;
