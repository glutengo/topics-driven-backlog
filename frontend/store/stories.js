import Vue from 'vue';

export default {
    namespaced: true,
    state: {
        initialised: false,
        STATUS: { OPEN: 0, IN_PROGRESS: 1, CLOSED: 2, CANCELLED: 3 },
        statusMap: [
            { name: 'open', css: 'badge-dark' },
            { name: 'in progress', css: 'badge-warning' },
            { name: 'closed', css: 'badge-success' },
            { name: 'cancelled', css: 'badge-danger' },
        ],
        data: [],
    },
    mutations: {
        /**
         * Set one or all stories
         * @param {object} state
         * @param {object} payload
         */
        set: (state, payload) => {
            if (payload.stories) {
                Vue.set(state, 'data', payload.stories);
            }

            if (payload.story) {
                const isPresent = state.data.find(
                    story => story.id === payload.story.id
                );

                if (isPresent) {
                    Vue.set(
                        state,
                        'data',
                        state.data.map(
                            story =>
                                story.id === payload.story.id
                                    ? payload.story
                                    : story
                        )
                    );
                } else {
                    state.data.push(payload.story);
                }
            }
        },

        /**
         * Remove story from data set
         * @param {object} state
         * @param {object} payload
         */
        remove: (state, payload) => {
            Vue.set(
                state,
                'data',
                state.data.filter(story => story.id !== payload.id)
            );
        },
    },
    actions: {
        /**
         * Initialises the stories store
         * @param {object} state
         * @param {function} dispatch
         */
        init({ state, dispatch }) {
            if (!state.initialised) {
                dispatch('fetch').then(() => {
                    state.initialised = true;
                });
            }
        },

        /**
         * Fetches all stories from the API and updates store
         * @param {function} commit
         * @returns {Promise}
         */
        fetch({ commit }) {
            return new Promise((resolve, reject) => {
                Vue.http.get('/stories').then(response => {
                    commit('set', {
                        stories: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Saves a story to the API and updates store
         * @param {function} commit
         * @param {object} payload (story)
         * @returns {Promise}
         */
        save({ commit }, payload) {
            return new Promise((resolve, reject) => {
                Vue.http.post('/stories', payload.story).then(response => {
                    commit('set', {
                        story: response.body,
                    });

                    resolve(response.body);
                }, reject);
            });
        },

        /**
         * Patches a given field of a given story with a given value or all given values
         * @param {function} commit
         * @param {object} payload (id, field, value, values)
         * @returns {Promise}
         */
        patch({ commit, dispatch }, payload) {
            return new Promise((resolve, reject) => {
                let values = {};

                if (payload.values) {
                    values = payload.values;
                } else {
                    values[payload.field] = payload.value;
                }

                Vue.http
                    .patch(`/stories/${payload.id}`, values)
                    .then(response => {
                        if (payload.fetch) {
                            dispatch('fetch').then(resolve, reject);
                        } else {
                            commit('set', {
                                story: response.body,
                            });

                            resolve(response.body);
                        }
                    }, reject);
            });
        },

        /**
         * Deletes a given story
         * @param {function} commit
         * @param {object} payload (id)
         * @returns {Promise}
         */
        delete({ commit }, payload) {
            return new Promise((resolve, reject) => {
                Vue.http.delete(`/stories/${payload.id}`).then(() => {
                    commit('remove', payload);

                    resolve();
                }, reject);
            });
        },
    },
    getters: {
        /**
         * All stories
         * @param {object} state
         * @returns {array}
         */
        all: state => (state.initialised ? state.data : []),

        /**
         * Find a story by identifier
         * @param {object} state
         * @returns {(object|null)}
         */
        byIdentifier: state => value =>
            state.initialised
                ? state.data.find(s => s.identifier === value)
                : null,

        /**
         * Find stories by field and value
         * @param {object} state
         * @returns {array}
         */
        find: state => (field, value) =>
            state.initialised ? state.data.filter(s => s[field] === value) : [],
    },
};
