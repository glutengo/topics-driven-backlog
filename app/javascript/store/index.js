import Vue from 'vue';
import Vuex from 'vuex';

import stories from './stories';

Vue.use(Vuex);

export default new Vuex.Store({
    modules: { stories },
    actions: {
        init({ dispatch }) {
            dispatch('stories/init');
        },
    }
});