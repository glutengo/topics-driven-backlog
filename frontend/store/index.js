import Vue from 'vue';
import Vuex from 'vuex';
import moment from 'moment';

import courses from '@/store/modules/courses';
import projects from '@/store/modules/projects';
import sprints from '@/store/modules/sprints';
import stories from '@/store/modules/stories';

Vue.use(Vuex);

export const state = {
    initialised: false,
    currentDate: moment().format('YYYY-MM-DD'),
};

export const actions = {
    /**
     * Initialise store and submodules
     * @param {object} state
     * @param {function} dispatch
     */
    init({ state, dispatch }) {
        dispatch('courses/init');
        state.initialised = true;
    },
};

export default new Vuex.Store({
    modules: { courses, projects, sprints, stories },
    state,
    actions,
});
