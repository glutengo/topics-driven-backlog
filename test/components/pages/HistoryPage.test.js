import { mount, createLocalVue } from 'vue-test-utils';
import { state } from '@/store/stories';
import Vuex from 'vuex';
import Vuelidate from 'vuelidate';
import HistoryPage from '@/components/pages/HistoryPage';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuelidate);

describe('HistoryPage.test.js', () => {
    let cmp, store, actions, storyGetters, courseGetters;

    beforeEach(() => {
        actions = {
            save: jest.fn(),
            patch: jest.fn(),
        };

        storyGetters = {
            all: () => [
                { id: 1, position: 2, status: 0 },
                { id: 2, position: 1, status: 0 },
                { id: 3, position: 3, status: 0, sprint_id: 1 },
                { id: 4, position: 4, status: 1 },
                { id: 5, position: 5, status: 0, sprint_id: 1 },
            ],
            find: () => () => [
                { id: 5, position: 5, status: 0, sprint_id: 1 },
                { id: 3, position: 3, status: 0, sprint_id: 1 },
            ],
        };

        courseGetters = {
            allSprints: () => () => [
                { id: 1, start_date: '2000-01-02' },
                { id: 2, start_date: '2000-01-01' },
            ],
        };

        store = new Vuex.Store({
            modules: {
                stories: {
                    namespaced: true,
                    state: {
                        STATUS: state.STATUS,
                        statusMap: state.statusMap,
                    },
                    actions,
                    getters: storyGetters,
                },
                courses: {
                    namespaced: true,
                    getters: courseGetters,
                },
            },
        });

        cmp = mount(HistoryPage, {
            localVue,
            store,
        });
    });

    it('has sorted stories in backlog', () => {
        expect(cmp.vm.backlog).toEqual([
            { id: 2, position: 1, status: 0 },
            { id: 1, position: 2, status: 0 },
            { id: 4, position: 4, status: 1 },
        ]);
    });

    it('has sorted sprints ', () => {
        expect(cmp.vm.sprints).toEqual([
            { id: 2, start_date: '2000-01-01' },
            { id: 1, start_date: '2000-01-02' },
        ]);
    });

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot();
    });
});
