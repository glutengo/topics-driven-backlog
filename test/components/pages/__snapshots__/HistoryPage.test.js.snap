// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryPage.test.js has the expected html structure 1`] = `
<section
  id="history-page"
>
  <h1>
    History
  </h1>
   
  <div>
    <h2>
      
    </h2>
     
    <table
      class="table table-striped"
    >
      <thead>
        <tr>
          <th>
            Identifier
          </th>
           
          <th>
            Story
          </th>
           
          <th>
            Story points
          </th>
           
          <th>
            Status
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr
          class="story-item"
        >
          <!---->
           
          <td>
            
            
        
          </td>
           
          <td>
            
            
        
          </td>
           
          <td>
            
            not estimated
        
          </td>
           
          <td>
            <span
              class="badge badge-dark"
            >
              
                open
            
            </span>
          </td>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </tr>
        <tr
          class="story-item"
        >
          <!---->
           
          <td>
            
            
        
          </td>
           
          <td>
            
            
        
          </td>
           
          <td>
            
            not estimated
        
          </td>
           
          <td>
            <span
              class="badge badge-dark"
            >
              
                open
            
            </span>
          </td>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </tr>
      </tbody>
    </table>
  </div>
  <div>
    <h2>
      
    </h2>
     
    <table
      class="table table-striped"
    >
      <thead>
        <tr>
          <th>
            Identifier
          </th>
           
          <th>
            Story
          </th>
           
          <th>
            Story points
          </th>
           
          <th>
            Status
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr
          class="story-item"
        >
          <!---->
           
          <td>
            
            
        
          </td>
           
          <td>
            
            
        
          </td>
           
          <td>
            
            not estimated
        
          </td>
           
          <td>
            <span
              class="badge badge-dark"
            >
              
                open
            
            </span>
          </td>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </tr>
        <tr
          class="story-item"
        >
          <!---->
           
          <td>
            
            
        
          </td>
           
          <td>
            
            
        
          </td>
           
          <td>
            
            not estimated
        
          </td>
           
          <td>
            <span
              class="badge badge-dark"
            >
              
                open
            
            </span>
          </td>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </tr>
      </tbody>
    </table>
  </div>
   
  <h2>
    Backlog
  </h2>
   
  <table
    class="table table-striped"
  >
    <thead>
      <tr>
        <th>
          Identifier
        </th>
         
        <th>
          Story
        </th>
         
        <th>
          Story points
        </th>
         
        <th>
          Status
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr
        class="story-item"
      >
        <!---->
         
        <td>
          
            
        
        </td>
         
        <td>
          
            
        
        </td>
         
        <td>
          
            not estimated
        
        </td>
         
        <td>
          <span
            class="badge badge-dark"
          >
            
                open
            
          </span>
        </td>
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
      </tr>
      <tr
        class="story-item"
      >
        <!---->
         
        <td>
          
            
        
        </td>
         
        <td>
          
            
        
        </td>
         
        <td>
          
            not estimated
        
        </td>
         
        <td>
          <span
            class="badge badge-dark"
          >
            
                open
            
          </span>
        </td>
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
      </tr>
      <tr
        class="story-item"
      >
        <!---->
         
        <td>
          
            
        
        </td>
         
        <td>
          
            
        
        </td>
         
        <td>
          
            not estimated
        
        </td>
         
        <td>
          <span
            class="badge badge-warning"
          >
            
                in progress
            
          </span>
        </td>
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
      </tr>
    </tbody>
  </table>
</section>
`;
